server/api.js

```diff
@@ -1,50 +1,70 @@
+                if(canConnectToDatabases(err, res)) {
+                    res.status(200).send('ok');
+                }
+            });
         });
 
         // create
         app.post('/api/todos/', function (req, res) {
             var todo = req.body;
             app.mySqlConnection(function (err, con) {
-                con.query('insert into todo_list set ?', todo, function (err, result) {
-                    con.release();
-                    res.status(201).send({id: result.insertId});
-                });
+                if(canConnectToDatabases(err, res)) {
+                    con.query('insert into todo_list set ?', todo, function (err, result) {
+                        con.release();
+                        res.status(201).send({id: result.insertId});
+                    });
+                }
             });
         });
 
         // retrieve
         app.get('/api/todos/', function (req, res) {
             app.mySqlConnection(function (err, con) {
-                con.query('select * from todo_list', function (err, result) {
-                    con.release();
-                    res.status(200).send(result);
-                });
+                if(canConnectToDatabases(err, res)) {
+                    con.query('select * from todo_list', function (err, result) {
+                        con.release();
+                        res.status(200).send(result);
+                    });
+                }
             });
         });
 
         // update
         app.put('/api/todos/:id', function (req, res) {
             var todo = req.body;
             app.mySqlConnection(function (err, con) {
-                con.query('update todo_list set ? where id = ?', [todo, todo.id], function (err, result) {
-                    con.release();
-                    res.status(200).send({id: result});
-                });
+                if(canConnectToDatabases(err, res)) {
+                    con.query('update todo_list set ? where id = ?', [todo, todo.id], function (err, result) {
+                        con.release();
+                        res.status(200).send({id: result});
+                    });
+                }
             });
         });
 
         // delete
         app.delete('/api/todos/:id/', function (req, res) {
             var id = req.params.id;
             app.mySqlConnection(function (err, con) {
-                con.query('delete from todo_list where id = ?', [id], function (err, result) {
-                    con.release();
-                    res.status(200).send({id: result});
-                });
+                if(canConnectToDatabases(err, res)) {
+                    con.query('delete from todo_list where id = ?', [id], function (err, result) {
+                        con.release();
+                        res.status(200).send({id: result});
+                    });
+                }
             });
         });
     }
 };
+
+function canConnectToDatabases(err, res) {
```

Comment:
Minor structural point but I think it'll lend clarity to the calling code: can you change each caller to 
```
if (!err) {
  con.query(...)
} else {
  sendError(res);
}
```
Where `sendError` is this method altered to just call `res.status(500).send(..)`.